gemini/use-cases/retrieval-augmented-generation/multimodal_rag_langchain.ipynb

```diff
@@ -0,0 +1,1026 @@
+    "    }\n",
+    "    | RunnableLambda(img_prompt_func)\n",
+    "    | ChatVertexAI(\n",
+    "        temperature=0, model_name=\"gemini-pro-vision\", max_output_tokens=1024\n",
+    "    )  # Multi-modal LLM\n",
+    "    | StrOutputParser()\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Process user query"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "query = \"\"\"\n",
+    " - What are the critical difference between various graphs for Class A Share?\n",
+    " - Which index best matches Class A share performance closely where Google is not already a part? Explain the reasoning.\n",
+    " - Identify key chart patterns for Google Class A shares.\n",
+    " - What is cost of revenues, operating expenses and net income for 2020. Do mention the percentage change\n",
+    " - What was the effect of Covid in the 2020 financial year?\n",
+    " - What are the total revenues for APAC and USA for 2021?\n",
+    " - What is deferred income taxes?\n",
+    " - How do you compute net income per share?\n",
+    " - What drove percentage change in the consolidated revenue and cost of revenue for the year 2021 and was there any effect of Covid?\n",
+    " - What is the cause of 41% increase in revenue from 2020 to 2021 and how much is dollar change?\n",
+    "\"\"\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Get Retrieved documents"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# List of source documents\n",
+    "docs = retriever_multi_vector_img.get_relevant_documents(query, limit=10)\n",
+    "\n",
+    "source_docs = split_image_text_types(docs)\n",
+    "\n",
+    "print(source_docs[\"texts\"])\n",
+    "\n",
+    "for i in source_docs[\"images\"]:\n",
+    "    display(Image(base64.b64decode(i)))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Get generative response"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "result = chain_multimodal_rag.invoke(query)\n",
+    "\n",
+    "Markdown(result)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "KwNrHCqbi3xi"
+   },
+   "source": [
+    "## Conclusions"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "05jynhZnkgxn"
+   },
+   "source": [
+    "Congratulations on making it through this multimodal RAG notebook!\n",
+    "\n",
+    "While multimodal RAG can be quite powerful, note that it can face some limitations:\n",
+    "\n",
+    "* **Data dependency:** Needs high-quality paired text and visuals.\n",
+    "* **Computationally demanding:** Processing multimodal data is resource-intensive.\n",
```

Comment:
Clarify if "resource-intensive" here is referring to the generation of multimodal embeddings at design-time or fetching/using them at run-time, or both?